src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp

```diff
@@ -215,6 +216,12 @@ int ExecuteManagedAssembly(
     coreClrDllPath.append("/");
     coreClrDllPath.append(coreClrDll);
 
+#if defined(__LINUX__)
+    std::string coreClrTPproviderDllPath(clrFilesAbsolutePath);
```

Comment:
I looked at the the Nuget Packaging logic, it packages all .so in the Folder level of libcoreclr.so so we should be good to go. I al so checked the created package and found this new shared object already in it